### ISSUE TITLE
Incorrect highlight python decorator

### ISSUE THREAD

**Issue by PavelBlend** [notepad-plus-plus/notepad-plus-plus#5894]
### Description of the Issue
The decorator color is not yellow if it contains a dot.

### Steps to Reproduce the Issue

Paste this code into Notepad ++ 7.7.1 32 bit:

```python
import test_module


@test_module.test_decorator
def test_function():
    pass
```

### Expected Behavior
The entire decorator should be yellow.

### Actual Behavior
![decorator_color_test](https://user-images.githubusercontent.com/7983249/60758137-6ecd7d80-a01c-11e9-8a1b-887df9bf3dc5.jpg)

**Comment by Ekopalypse** [comment]
npp uses scintilla as its core lexing component and therefore issue needs to be reported [here](https://sourceforge.net/p/scintilla/bugs/).
From my limited understanding it seems the issue is [here](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/scintilla/lexers/LexPython.cxx#L666).
If I replace this with `if (!IsAWordChar(sc.ch, options.unicodeIdentifiers)) {` then it seems to work.

**Comment by PavelBlend** [comment]
Created a task:
https://sourceforge.net/p/scintilla/bugs/2139/

**Comment by PavelBlend** [comment]
I wrote an issue to lexilla and got the following response:
https://github.com/ScintillaOrg/lexilla/pull/219#issuecomment-1849782847

Is it possible to fix this in Notepad++?

**Comment by rdipardo** [comment]
@PavelBlend

Like I [tried to explain][0] before, there are two different lexer properties to configure.

The function decorator is in the `SCE_P_DECORATOR` state (`id=15`), which already has a style associated with it by the default Notepad++ theme.

The *method* decorator is in the `SCE_P_ATTRIBUTE` state (`id=20`), which does not yet have any style associated with it.

Here's how it should look in [SciTE](https://www.scintilla.org/SciTEDownload.html) with both options configured:

    ; %UserProfile%\SciTEUser.properties

    lexer.python.decorator.attributes=1
    lexer.python.identifier.attributes=1

    ; decorator style
    style.python.15=fore:#FF8000,italics,bold

    ; attribute style
    style.python.20=fore:#CD0000,italics,bold,underlined

![npp-gh-issue-5894-gscite-542](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/59004801/55226957-61cd-4a6c-8628-d852bb388894)

[0]: https://github.com/notepad-plus-plus/notepad-plus-plus/pull/14507#issuecomment-1866828226

**Comment by PavelBlend** [comment]
@donho I expected different behavior. I have a standard theme. Decorators look like this:
![01](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/7983249/f0bec812-7a2a-4158-b054-2c591b832046)

<br><br/>

I added a new `ATTRIBUTES` color to the `stylers.xml `file. This is the result:
![02](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/7983249/d2a925b9-af65-4679-a70e-d0624831c113)

<br><br/>

I would like to have such a result:
![2](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/7983249/034a87b3-a893-48c1-aa82-e185f9332a38)

<br><br/>

Is it possible to specify the color of decorator attributes separately from other attributes?

**Comment by rdipardo** [comment]
@PavelBlend 

> I added a new `ATTRIBUTES` color to the `stylers.xml `file. This is the result: ![02](https://private-user-images.githubusercontent.com/7983249/320074736-d2a925b9-af65-4679-a70e-d0624831c113.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA1NDMzMzksIm5iZiI6MTc0MDU0MzAzOSwicGF0aCI6Ii83OTgzMjQ5LzMyMDA3NDczNi1kMmE5MjViOS1hZjY1LTQ2NzktYTcwZS1kMDYyNDgzMWMxMTMuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDIyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAyMjZUMDQxMDM5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MGRhYWZjNTI2ZTJhOTcwNzdkNWM4Y2ZjZjJmNWRlZjQ1NGY2MGY4MzEwMTY3MzA0OTBiMzhhZGNiYjdkY2Q1NyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.mqntUHoxDzcfr_KBlhSekjXTQQcxI6woHBV70VDnLhw)
> 

This has finally been fixed in Lexilla: ScintillaOrg/lexilla@df5fefe (not merged with Notepad++ yet...maybe in 8.7.9...?)

    ; decorator style
    style.python.15=fore:#FF8000,italics,bold

    lexer.python.decorator.attributes=1

    ; not needed for decorators anymore so can be disabled
    ; lexer.python.identifier.attributes=1

![scite555-python-decorators](https://github.com/user-attachments/assets/5eb576fd-f74c-46d2-97cb-abeeee8c3a36)
